tool/tsh/gcp.go

```diff
@@ -117,15 +158,68 @@ func (a *gcpApp) StartLocalProxies() error {
 // Close makes all necessary close calls.
 func (a *gcpApp) Close() error {
 	var errs []error
+	// close proxies
 	if a.localALPNProxy != nil {
 		errs = append(errs, a.localALPNProxy.Close())
 	}
 	if a.localForwardProxy != nil {
 		errs = append(errs, a.localForwardProxy.Close())
 	}
+	// remove boto config
+	errs = append(errs, a.removeBotoConfig()...)
 	return trace.NewAggregate(errs...)
 }
 
+// removeBotoConfig removes config files written by WriteBotoConfig.
+func (a *gcpApp) removeBotoConfig() []error {
+	// try to remove both files
+	return []error{os.Remove(a.getExternalAccountFilePath()), os.Remove(a.getBotoConfigPath())}
+}
+
+func (a *gcpApp) getBotoConfigDir() string {
+	return path.Join(profile.FullProfilePath(a.cf.HomePath), "gcp", a.app.ClusterName, a.app.Name)
+}
+
+func (a *gcpApp) getBotoConfigPath() string {
+	return path.Join(a.getBotoConfigDir(), a.prefix+"_boto.cfg")
+}
+
+func (a *gcpApp) getExternalAccountFilePath() string {
+	return path.Join(a.getBotoConfigDir(), a.prefix+"_external.json")
+}
+
+// getBotoConfig returns minimal boto configuration, referencing an external account file.
+func (a *gcpApp) getBotoConfig() string {
+	return fmt.Sprintf(`[Credentials]
+gs_external_account_authorized_user_file = %v
+`, a.getExternalAccountFilePath())
+}
+
+// getExternalAccountFile returns the contents of external account file, which depend on a current secret.
+func (a *gcpApp) getExternalAccountFile() string {
+	return fmt.Sprintf(`{ "type": "external_account_authorized_user","token": %q }`, a.secret)
```

Comment:
It would be great if we have a reference link for this.

Also, we only need the secret here? How about the CA cert? `gsutil` can read the CA cert path from env var?

Reply:
Yep, it will read the CA cert path from env var.